reactor-pool/src/main/java/reactor/pool/SimpleDequePool.java

```diff
@@ -99,6 +99,9 @@ public class SimpleDequePool<POOLABLE> extends AbstractPool<POOLABLE> {
 
 	Disposable evictionTask;
 
+	// Flag used to avoid creating resources while warmup is in progress
+	volatile boolean warmupInProgress = false;
```

Comment:
Would it make sense to prevent the `drainLoop()` being called concurrently instead? Let's say that > 1 concurrent acquires happen. Both increment the demand and add themselves to the `Borrower` queue. However, just one enters the `drainLoop` and delivers all necessary resources. A loop can check if more demand has been added before it exits in order to repeat the procedure and ensure every acquire is satisfied in the end. But that would guarantee no simultaneous warmup happens.

Reply:
Just to clarify why drainLoop can be executed concurrently: both evictInBackground and pendingOffer methods have a check like this:

				if (WIP.decrementAndGet(this) > 0) {
					drainLoop();
				}
				
And it doesn't exclude other threads from entering.

Reply:
if I'm correct `drainLoop` can't be run concurrently, it is protected using the WIP design pattern.
However, I will need to give more thinking about your comment, thanks.

Reply:
This is not a typical WIP pattern that is in use here. The above condition should be as the one in the `drain` method:

	void drain() {
		if (WIP.getAndIncrement(this) == 0) {
			drainLoop();
		}
	}
	
but it's not the case for `evictInBackground` and `pendingOffer`.